tests/crypto/src/common_test.h

```diff
@@ -0,0 +1,452 @@
+	const char                      *p_prk;                         /**< Pointer to hkdf PseudoRandom Key in hex string format. */
+	const char                      *p_salt;                        /**< Pointer to hkdf salt in hex string format. */
+	const char                      *p_info;                        /**< Pointer to hkdf optional application specific information in hex string format. */
+} test_vector_hkdf_t;
+
+const char *get_vector_name(const test_case_t *tc, uint32_t v);
+uint32_t get_vector_count(const test_case_t *tc);
+
+
+/* This context will be populated by the below initialization function.
+ * If the initialization is successful it can be used in subsequent calls to
+ * mbedtls drbg APIs.
+ */
+extern mbedtls_ctr_drbg_context ctr_drbg_ctx;
+
+/**@brief Function for initializing counter mode deterministic random byte generator.
+ *
+ * @details Should only be called once.
+ *
+ * @param[in]		p_optional_seed		Pointer to a seed. Can be NULL. In that case uses internal seed.
+ * @param[in]      	len					Length of seed in bytes. Ignored if seed is NULL.
+ *
+ * @return								0 on success, else an error code.
+ *
+ *
+ *
+ *
+ *
+ */
+int init_ctr_drbg(const unsigned char *p_optional_seed, size_t len);
+
+/**@brief Function for converting hex string to bytes.
+ *
+ * @param[in,out]  p_output     Pointer to the output buffer where the bytes will be stored.
+ * @param[in]      p_input      Pointer to the input hexstring.
+ *
+ * @return                    The number of bytes that was read and converted.
+ */
+uint32_t unhexify(uint8_t *p_output, char const *p_input);
+
+
+/**@brief Function for setting LED_1 to indicate a start of time measurement.
+ *
+ */
+void start_time_measurement(void);
+
+
+/**@brief Function for clearing LED_1 to indicate a stop of time measurement.
+ *
+ */
+void stop_time_measurement(void);
+
+/**@brief Macro(s) for decorating test vector names with file and line information.
+ *
+ */
+#define _TV_DECORATE2(file, line, name)     name " -- " "[" file ":" #line "]"
+#define _TV_DECORATE1(file, line, name)     _TV_DECORATE2(file, line, name)
+#define TV_NAME(name)                       _TV_DECORATE1(__FILE__, __LINE__, name)
+
+/**
+ * @brief Macro for concatenating two tokens in macro expansion.
+ *
+ * @note This macro is expanded in two steps so that tokens given as macros
+ *       themselves are fully expanded before they are merged.
+ *
+ * @param[in] p1 First token.
+ * @param[in] p2 Second token.
+ *
+ * @return The two tokens merged into one, unless they cannot together form
+ *         a valid token (in such case, the preprocessor issues a warning and
+ *         does not perform the concatenation).
+ *
+ * @sa NRFX_CONCAT_3
+ */
+#define CONCAT_2(p1, p2)       CONCAT_2_(p1, p2)
+
+
+/** @brief Internal macro used by @ref CONCAT_2 to perform the expansion in two steps. */
+#define CONCAT_2_(p1, p2)      p1 ## p2
```

Comment:
We already have this, see zephyr/include/misc/util.h

Reply:
Done in 1ce8d33